freqtrade/rpc/discord.py

```diff
@@ -0,0 +1,108 @@
+            open_date = msg.get('open_date').strftime('%Y-%m-%d %H:%M:%S')
+            close_date = msg.get('close_date').strftime(
+                '%Y-%m-%d %H:%M:%S') if msg.get('close_date') else ''
+
+            embeds = [{
+                'title': '{} Trade: {}'.format(
+                    'Profit' if profit_ratio > 0 else 'Loss',
+                    msg.get('pair')),
+                'color': (0x00FF00 if profit_ratio > 0 else 0xFF0000),
+                'fields': [
+                    {'name': 'Trade ID', 'value': msg.get('id'), 'inline': True},
+                    {'name': 'Exchange', 'value': msg.get('exchange').capitalize(), 'inline': True},
+                    {'name': 'Pair', 'value': msg.get('pair'), 'inline': True},
+                    {'name': 'Direction', 'value': 'Short' if msg.get(
+                        'is_short') else 'Long', 'inline': True},
+                    {'name': 'Open rate', 'value': msg.get('open_rate'), 'inline': True},
+                    {'name': 'Close rate', 'value': msg.get('close_rate'), 'inline': True},
+                    {'name': 'Amount', 'value': msg.get('amount'), 'inline': True},
+                    {'name': 'Open order', 'value': msg.get('open_order_id'), 'inline': True},
+                    {'name': 'Open date', 'value': open_date, 'inline': True},
+                    {'name': 'Close date', 'value': close_date, 'inline': True},
+                    {'name': 'Profit', 'value': msg.get('profit_amount'), 'inline': True},
+                    {'name': 'Profitability', 'value': '{:.2f}%'.format(
+                        profit_ratio * 100), 'inline': True},
+                    {'name': 'Stake currency', 'value': msg.get('stake_currency'), 'inline': True},
+                    {'name': 'Fiat currency', 'value': msg.get(
+                        'fiat_display_currency'), 'inline': True},
+                    {'name': 'Buy Tag', 'value': msg.get('enter_tag'), 'inline': True},
+                    {'name': 'Sell Reason', 'value': msg.get('exit_reason'), 'inline': True},
+                    {'name': 'Strategy', 'value': self.strategy, 'inline': True},
+                    {'name': 'Timeframe', 'value': self.timeframe, 'inline': True},
+                ],
+            }]
+
+            # convert all value in fields to string for discord
+            for embed in embeds:
+                for field in embed['fields']:
+                    field['value'] = str(field['value'])
+
+            # Send the message to discord channel
+            payload = {
+                'embeds': embeds,
+            }
+            headers = {
+                'Content-Type': 'application/json',
+            }
+            try:
+                requests.post(
+                    self.config['discord']['webhook_url'],
+                    data=json.dumps(payload),
+                    headers=headers)
+            except Exception as e:
+                self.logger.error(f"Failed to send discord message: {e}")
```

Comment:
you may want to slightly refactor this - splitting message generation from sending the message (basically using a similar pattern than the webhook thing).

you'll see that it will make this simpler to test - as you'll just get "embeds" (or payload?)  as parameter - and can assert if it's content is what you expect.

You may even be able to subclass `Webhook` instead (this really only provides a better way to configure the webhooks) - and use it's `_send_msg()` method for the actual network sending.